internal/server/middleware.go

```diff
@@ -182,3 +182,76 @@ func getCookie(req *http.Request, name string) (string, error) {
 	}
 	return url.QueryUnescape(cookie.Value)
 }
+
+func OrganizationFromDomain(defaultOrgName, defaultOrgDomain string) gin.HandlerFunc {
```

Comment:
I suspect writing this as a regular function (not a middleware) would remove the need for the `goto` and make it easier to follow:

```go
func organizationFromRequest(req *http.Request, defaultOrgName string) (*models.Organization, error) {
   ...
}
```